File: lvl.Oidc.AuthorizationServer/Models/Scope.cs
using FluentNHibernate.Automapping;
using FluentNHibernate.Automapping.Alterations;
using IdentityServer4.Models;
using lvl.Ontology;
using System.ComponentModel;

namespace lvl.Oidc.AuthorizationServer.Models
{
    [Description(nameof(Scope))]
    public class ScopeEntity : Scope, IEntity
    {
        public int Id { get; set; }
    }

    public class ScopeOverride : IAutoMappingOverride<Scope>
    {
        public void Override(AutoMapping<Scope> mapping)
        {
            mapping.Id(scope => scope.Name);

            mapping.HasMany(scope => scope.UserClaims).Element("Value");
        }
    }
}

File: @lvl.oidc/src/models/scope.ts
import { IEntity } from '@lvl/core';

export class Scope implements IEntity {
    public id: number;
    public name: string;
    public displayName: string;
    public description: string;
    public required: boolean;
    public emphasize: boolean;
    public showInDiscoveryDocument: boolean;
    public userClaims: string[];
}
